frontend/src/router/index.ts

```diff
@@ -1,30 +1,72 @@
-import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router'
-import HomeView from '../views/HomeView.vue'
+import { MutationTypes } from '@/store/mutation-types';
+import { store } from '@/store/store';
+import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router';
+import HomeView from '../views/HomeView.vue';
 
 const routes: Array<RouteRecordRaw> = [
-  {
-    path: '/',
-    name: 'home',
-    component: HomeView
-  },
-  {
-    path: '/about',
-    name: 'about',
-    // route level code-splitting
-    // this generates a separate chunk (about.[hash].js) for this route
-    // which is lazy-loaded when the route is visited.
-    component: () => import(/* webpackChunkName: "about" */ '../views/AboutView.vue')
-  },
-  {
-    path: '/signup',
-    name: 'signup',
-    component: () => import('../views/SignUpView.vue')
-  }
-]
+    {
+        path: '/',
+        name: 'home',
+        component: HomeView,
+    },
+    {
+        path: '/about',
+        name: 'about',
+        // route level code-splitting
+        // this generates a separate chunk (about.[hash].js) for this route
+        // which is lazy-loaded when the route is visited.
+        component: () =>
+            import(/* webpackChunkName: 'about' */ '../views/AboutView.vue'),
+    },
+    {
+        path: '/signup',
+        name: 'signup',
+        component: () => import('../views/SignUpView.vue'),
+    },
+    {
+        path: '/wallet',
+        name: 'wallet',
+        component: () => import('../views/WalletView.vue'),
+    },
+    {
+        path: '/assets',
+        name: 'assets',
+        component: () => import('../views/AssetsView.vue'),
+    },
+    {
+        path: '/logout',
+        name: 'logout',
+        component: () => import('../views/LogoutView.vue'),
+        meta: {
+            hideNavBar: true,
+        },
+    },
+];
 
 const router = createRouter({
-  history: createWebHistory(process.env.BASE_URL),
-  routes
-})
+    history: createWebHistory(process.env.BASE_URL),
+    routes,
+});
 
-export default router
+router.beforeEach((to, from, next) => {
+    const token = sessionStorage.getItem('access_token'); 
+    if (token) {
+        const endpoint = '/auth/validate';
+        fetch(process.env.VUE_APP_API_URL + endpoint, {
+            method: 'POST',
+            headers: {
+                'Authorization': `Bearer ${token}`
+            }
+        }).then(response => {
+            if (response.ok) {
+                next();
+            } else {
+                localStorage.removeItem('token');
+                return {
+                    name: 'NotFound'
+                };
```

Comment:
Ideally, we must always use the `next()` function inside the router beforeEach. If the user does not have a token or it is invalid we should redirect them to the login page using `next({ name: 'home' })`

Reply:
That was the thing I was struggling today. Because I wanted to check the token or the state `isLogged` and not return to the home screen.

But I was trying to understand how does VUE manage this.

Especially if the user IS VALID and how to proceed? `next(to)`?

Reply:
The `beforeEach` will be executed **before** the user goes to any route, so what you do here must always send the user to some route. If you call `next()` without params it will continue to the next route. If you wanna send the user to another route depending on some condition, you can call the `next` using the route params, like `next({ name: 'foo', params: { bar: true } })`. 

In this case, if the user is valid you can just use `next()`.